src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4821,6 +4824,33 @@ jint os::init_2(void) {
     }
   }
 
+  // Check the availability of MADV_POPULATE_WRITE.
+  if (UseMadvPopulateWrite) {
+    // Some downstream kernels recognize MADV_POPULATE_WRITE_value as another
+    // advice, so the check of versions is required here.
+    // See https://github.com/oracle/linux-uek/issues/23
+    struct utsname buffer;
+    int ret = uname(&buffer);
+    if (ret != 0) {
+      log_warning(os)("uname(2) failed to get kernel version: %s",
+                      os::errno_name(ret));
+    }
+
+    long major = -1, minor = -1;
+    bool supportMadvPopulateWrite =
+      (ret == 0 &&
+       (sscanf(buffer.release, "%ld.%ld", &major, &minor) == 2) &&
+       (major > 5 || (major == 5 && minor >= 14)) &&
```

Comment:
Why isn't this using the `os::Linux::kernel_version()` function?

Reply:
Just not get the idea. Previous discussions mentioned the problems of kernel_version, so I used an alternative method. Using kernel_version here is fine with me.

Reply:
FYI: @jdksjolen is going to fix `kernel_version` with #18697.